x/evm/events.go

```diff
@@ -1,21 +1,106 @@
 // Copyright (c) 2023-2024 Nibi, Inc.
 package evm
 
+import (
+	"fmt"
+	"strconv"
+
+	"cosmossdk.io/errors"
+	abci "github.com/cometbft/cometbft/abci/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
 // Evm module events
 const (
-	EventTypeEthereumTx = TypeMsgEthereumTx
-	EventTypeBlockBloom = "block_bloom"
-	EventTypeTxLog      = "tx_log"
-
-	AttributeKeyRecipient      = "recipient"
-	AttributeKeyTxHash         = "txHash"
-	AttributeKeyEthereumTxHash = "ethereumTxHash"
-	AttributeKeyTxIndex        = "txIndex"
-	AttributeKeyTxGasUsed      = "txGasUsed"
-	AttributeKeyTxType         = "txType"
-	AttributeKeyTxLog          = "txLog"
-	// tx failed in eth vm execution
-	AttributeKeyEthereumTxFailed = "ethereumTxFailed"
-	AttributeValueCategory       = ModuleName
-	AttributeKeyEthereumBloom    = "bloom"
+	// proto.MessageName(new(evm.EventBlockBloom))
+	TypeUrlEventBlockBloom = "eth.evm.v1.EventBlockBloom"
+
+	// proto.MessageName(new(evm.EventTxLog))
+	TypeUrlEventTxLog = "eth.evm.v1.EventTxLog"
+
+	// proto.MessageName(new(evm.TypeUrlEventEthereumTx))
+	TypeUrlEventEthereumTx = "eth.evm.v1.EventEthereumTx"
+
+	// Untyped events and attribuges
+
+	// Used in non-typed event "message"
+	MessageEventAttrTxType = "tx_type"
+
+	// Used in non-typed event "pending_ethereum_tx"
+	PendingEthereumTxEvent            = "pending_ethereum_tx"
+	PendingEthereumTxEventAttrEthHash = "eth_hash"
+	PendingEthereumTxEventTxAttrIndex = "index"
```

Comment:
```suggestion
	PendingEthereumTxEventAttrIndex = "index"
```

To follow the convention of the others